### PR TITLE
Parameterized operators

### PR DESCRIPTION
**Pull request by bksaiki**
This PR adds `binary64` and `binary32` operators under the `define-language!` macro to replace the previous `real` / FPCore operators and constants. Herbie now parameterizes operators and constants (e.g. `sin -> sin.f64` for binary64) to be representation specific (see uwplse/herbie#319). The list of operators is now annoyingly long. This should be fixed soon.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -186,7 +187,6 @@ fn find_extracted(runner: &Runner, id: u32) -> &Extracted {
         .find(|(i, _)| runner.egraph.find(*i) == id)
         .map(|(_, ext)| ext)
         .expect("Couldn't find matching extraction!")
-        .clone()
```
Comment (pavpanchekha) [inline src/lib.rs:189]: Why don't we need a clone here?

File: src/math.rs [hunk 2]
```diff
@@ -42,102 +42,163 @@ impl IterationData<Math, ConstantFold> for IterData {
...
         "LN2" = Ln2,
         "LN10" = Ln10,
-        "PI" = Pi,
         "PI_2" = Pi2,
         "PI_4" = Pi4,
         "1_PI" = Pi1Alt,
         "2_PI" = Pi2Alt,
         "2_SQRTPI" = Sqrtpi2,
         "SQRT2" = Sqrt2,
         "SQRT1_2" = Sqrt1_2,
```
Comment (pavpanchekha) [inline src/math.rs]: Should we delete the rest of these too?

**Review by pavpanchekha — APPROVED** [review]

**Comment by pavpanchekha** [comment]
Perhaps we should make the existing operators also work? This way you could use new-Egg-Herbie with old-Herbie.

File: src/lib.rs (hunk 1 above)
Comment (bksaiki) [reply src/lib.rs:189]: I'm honestly not 100% sure. `cargo test` was failing if `.clone()` was left in (CI didn't fail though). I think `.clone()` in this context returns a double reference `&&Extracted` while the function is of type `&Extracted`.

**Comment by bksaiki** [comment]
Oh duh. Backwards compatibility is a good idea.

File: src/lib.rs (hunk 1 above)
Comment (pavpanchekha) [reply src/lib.rs:189]: @mwillsey could you help?

File: src/math.rs (hunk 2 above)
Comment (bksaiki) [reply src/math.rs]: If Herbie has no intention of using these constants, it makes sense to remove them.

File: src/lib.rs (hunk 1 above)
Comment (mwillsey) [reply src/lib.rs:189]: I'm a little confused about the confusion. If the compiler lets you not clone, you don't need to clone.

Comment (pavpanchekha) [reply src/lib.rs:189]: Hash tag rust. Sorry, I don't have rust intuition, only C++ intuition, and in C++, if someone removes a copy, you interrogate them viciously.

File: src/math.rs (hunk 2 above)
Comment (pavpanchekha) [reply src/math.rs]: Agreed.

**Comment by bksaiki** [comment]
Fixed

**Pull request merged**